octopus-server/src/main/java/octopus/teamcity/server/OctopusMetadataBuildStartProcessor.java

```diff
@@ -61,7 +61,9 @@ public void updateParameters(@NotNull BuildStartContext buildStartContext) {
 
         buildStartContext.addSharedParameter("commits", jsonData);
         buildStartContext.addSharedParameter("vcstype", vcsType);
-        buildStartContext.addSharedParameter("vcsroot", vcsRootUrl);
+        if (vcsRootUrl != null) {
+            buildStartContext.addSharedParameter("vcsroot", vcsRootUrl);

```

Comment:
I assume the `.get` method handles missing parameters since we retrieve that parameter in a different part of the plugin?

Reply:
Yes, the get method hands back a null if the value doesn't exist in the parameters, and doesn't cause any issues